src/frontend/mosh-server.cc

```diff
@@ -545,14 +572,26 @@ static void serve( int host_fd, Terminal::Complete &terminal, ServerConnection &
 
   while ( 1 ) {
     try {
+      static const uint64_t timeout_if_no_client = 60000;
+      int timeout = INT_MAX;
       uint64_t now = Network::timestamp();
 
-      const int timeout_if_no_client = 60000;
-      int timeout = min( network.wait_time(), terminal.wait_time( now ) );
+      timeout = min( timeout, network.wait_time() );
+      timeout = min( timeout, terminal.wait_time( now ) );
       if ( (!network.get_remote_state_num())
 	   || network.shutdown_in_progress() ) {
         timeout = min( timeout, 5000 );
       }
+      /* The server goes completely asleep if it has no remote peer. */
+      if ( network_timeout_ms ) {
+	int64_t wait = network_timeout_ms - ( now - network.get_latest_remote_state().timestamp );
```

Comment:
`wait` is already a function (in `sys/wait.h`). It doesn't cause problems currently, but it might be better to call this `wait_ms`.